src/FbTk/TextBox.cc

```diff
@@ -591,7 +591,7 @@ void TextBox::adjustPos(){
 }
 
 
-void TextBox::select(std::string::size_type pos, int length)
+void TextBox::select(std::string::size_type pos, unsigned int length)
```

Comment:
The very next line has and handles ```if (length < 0) {``` because that's a valid case (rtl selection)
I'll say that I generally feel very uneasy about random signage changes, because they can get you into really weird trouble when the range overflows and instead of "-1" you end w/ UINT_MAX as a temporary calculation, throwing your math completely off whack

Reply:
i tend to use the signed types these days … some of the code places in fluxbox i would rather change to "almost always" use signed ints over the unsigned flavours.

Reply:
That's fine as long as you pay attention with arithmetics, but you can't just assume that it won't matter and esp. if the change is driven by a compiler warning about signed/unsigned mixing, changes towards unsigned need to be completely vetted. Doing this blindly to silence the compiler is a recipe for disaster.

In this particular case it would also require a change of the functions signature to flag the direction .

Reply:
Moot because of separation.